packages/grafana-runtime/src/services/pluginExtensions/usePluginFunctions.ts

```diff
@@ -0,0 +1,20 @@
+import { UsePluginFunctionsOptions, UsePluginFunctionsResult } from './getPluginExtensions';
+
+export type UsePluginFunctions<T> = (options: UsePluginFunctionsOptions) => UsePluginFunctionsResult<T>;
+
+let singleton: UsePluginFunctions<unknown> | undefined;
+
+export function setPluginFunctionsHook(hook: UsePluginFunctions<unknown>): void {
+  // We allow overriding the registry in tests
+  if (singleton && process.env.NODE_ENV !== 'test') {
+    throw new Error('setUsePluginFunctionsHook() function should only be called once, when Grafana is starting.');
+  }
+  singleton = hook;
+}
+
+export function usePluginFunctions<T>(options: UsePluginFunctionsOptions): UsePluginFunctionsResult<T> {
+  if (!singleton) {
+    throw new Error('usePluginFunctions(options) can only be used after the Grafana instance has started.');
+  }
+  return singleton(options) as UsePluginFunctionsResult<T>;
```

Comment:
Not sure how to get rid of the type assertion here (and in the implementation)

Reply:
There is the `satisfies` keyword but it doesn't work here since we have `unknown` in the singleton. I don't think we can get rid of it so lets just keep it as it is.